src/stdlib_bitset_large.fypp

```diff
@@ -0,0 +1,1372 @@
+
+        start_bit = max( 0, start_pos )
+        end_bit   = min( stop_pos , self % num_bits-1 )
+        if ( end_bit < start_bit ) return
+
+        first_block = start_bit / block_size + 1
+        last_block  = end_bit / block_size + 1
+        if (first_block == last_block) then
+!         FIRST and LAST are in the same block
+            call mvbits( not(self % blocks(first_block)),        &
+                         start_bit - (first_block-1)*block_size, &
+                         end_bit - start_bit + 1,                &
+                         self % blocks(first_block),             &
+                         start_bit - (first_block-1)*block_size )
+            return
+        end if
+
+!     Do "partial" black containing FIRST
+        bit = start_bit - (first_block-1)*block_size
+        call mvbits( not(self % blocks(first_block) ), &
+                     bit,                              &
+                     block_size - bit,                 &
+                     self % blocks(first_block),       &
+                     bit )
+
+!     Do "partial" black containing LAST
+        bit = end_bit - (last_block-1)*block_size
+        call mvbits( not( self % blocks(last_block) ), &
+                     0,                                &
+                     bit+1,                            &
+                     self % blocks(last_block),        &
+                     0 )
+
+!     Do remaining blocks
+        do block_ = first_block+1, last_block-1
+            self % blocks(block_) = not( self % blocks(block_) )
+        end do
+
+    end subroutine flip_range_large
+
+    module subroutine from_string_large(self, string, status)
+!     Initializes the bitset `self` treating `string` as a binary literal
+!     `status` may have the values:
+!         `success` - if no problems were found,
+!         `alloc_fault` - if allocation of the bitset failed
+!         `char_string_too_large_error` - if `string` was too large, or
+!         `char_string_invalid_error` - if string had an invalid character.
+        class(bitset_large), intent(out) :: self
+        character(*), intent(in)         :: string
+        integer, intent(out), optional   :: status
+
+        character(*), parameter :: procedure = 'FROM_STRING'
+        integer(bits_kind) :: bit
+        integer(int64)     :: bits
+        character(1)       :: char
+
+        bits = len(string, kind=int64)
+        if ( bits > huge(0_bits_kind) ) then
+            call error_handler( 'STRING was too long for a ' //      &
+                                'BITSET_LARGE SELF.',                &
+                                char_string_too_large_error, status, &
+                                module_name, procedure )
+            return
+        end if
+
+        call init_zero_large( self, int(bits, kind=bits_kind), status )
+
+        if ( present(status) ) then
+            if ( status /= success ) return
+        end if
+
+        do bit = 1_bits_kind, bits
```

Comment:
Here there's a similar warning, this time the other way around:
```
/home/ian/stdlib/src/stdlib_bitset_large.f90:562:29:

  562 |         do bit = 1_bits_kind, bits
      |                             1
Warning: Possible change of value in conversion from INTEGER(8) to INTEGER(4) at (1) [-Wconversion]
```
I believe `bits` and `bit` should have the same kind, other than `bits_kind`, and since they are iterators there's no problem using `int64` as kind here.